src/style-spec/expression/definitions/within.js

```diff
@@ -160,32 +162,78 @@ function getTilePolygons(coordinates, bbox, canonical) {
     return polygons;
 }
 
-function pointsWithinPolygons(ctx: EvaluationContext, polygonGeometry: GeoJSONPolygons) {
-    const pointBBox = [Infinity, Infinity, -Infinity, -Infinity];
-    const polyBBox = [Infinity, Infinity, -Infinity, -Infinity];
-    const canonical = ctx.canonicalID();
+function updatePoint(p, bbox, polyBBox, worldSize) {
+    if (p[0] < polyBBox[0] || p[0] > polyBBox[2]) {
+        let shift = (p[0] - polyBBox[0] > worldSize / 2) ? -worldSize : (polyBBox[0] - p[0] > worldSize / 2) ? worldSize : 0;
+        if (shift === 0) {
+            shift = (p[0] - polyBBox[2] > worldSize / 2) ? -worldSize : (polyBBox[2] - p[0] > worldSize / 2) ? worldSize : 0;
+        }
+        p[0] += shift;
```

Comment:
Right, that's something I am wondering how to resolve, any suggestion? Should I use a while loop to make sure the geometries are finally unwrapped in case they are wrapped multiple times with multiple 180th?

Reply:
It depends what we want to tolerate as input data for the longitude ranges, as it's up to the implementation to see what is acceptable: https://tools.ietf.org/html/rfc7946#section-3.1.9. Another solution would be to do an early clipping of the geometry, and always wrap in `getTileCoordinates` as well as this function but it adds some complexity.

If you want to keep it the way you have it here, you could find how many _wraps_ away you are to determine the shift count. Here's a suggestion from this commit https://github.com/mapbox/mapbox-gl-js/commit/5b9e550b97494ec03f94d5f769b248bb5fffcdc6

```js
function updatePoint(p, bbox, polyBBox, worldSize) {
    if (p[0] < polyBBox[0] || p[0] > polyBBox[2]) {
        const halfWorldSize = worldSize * 0.5;
        const d0 = Math.max(p[0] - polyBBox[0], p[0] - polyBBox[2]);
        const d1 = Math.max(polyBBox[0] - p[0], polyBBox[2] - p[0]);
        if (d0 > halfWorldSize) p[0] -= Math.round(d0 / halfWorldSize) * halfWorldSize;
        if (d1 > halfWorldSize) p[0] += Math.round(d1 / halfWorldSize) * halfWorldSize;
    }
    updateBBox(bbox, p);
}
```

Reply:
Thank you very much! 🙇‍♀ 
From my understanding, `p[0] -= Math.round(d0 / halfWorldSize) * halfWorldSize;` the multiplier should be `worldSize` instead of `halfWorldSize`, only by shifting worldSize, the point would overlap with itself, otherwise, we would change the point's original position.

Taking this point [-3, 62.5] as example, the shifted result should be [-363, 62.5] instead of [-183, 62.5].

Reply:
@karimnaaji we never supported data that goes beyond one world wrap (which needs a single shift) — for example, [here's the code for it in GeoJSON-VT](https://github.com/mapbox/geojson-vt/blob/master/src/wrap.js), and I never saw any issues reported about it. So I don't think we should look too much into it with the `within` expression as well.

Reply:
As long as we're consistent it's ok to leave it as it is then.